packages/components/input/src/InputGroup.tsx

```diff
@@ -1,83 +1,61 @@
+/* eslint-disable complexity */
+import { useFormFieldState } from '@spark-ui/form-field'
 import {
   Children,
+  cloneElement,
   ComponentPropsWithoutRef,
   FC,
   forwardRef,
   isValidElement,
   PropsWithChildren,
   ReactElement,
   useMemo,
-  useState,
 } from 'react'
 
+import { InputContainer, InputContainerProps } from './InputContainer'
 import { inputGroupStyles, InputGroupStylesProps } from './InputGroup.styles'
 import { InputGroupContext } from './InputGroupContext'
-export interface InputGroupProps
-  extends ComponentPropsWithoutRef<'div'>,
-    Omit<InputGroupStylesProps, 'isFocused'> {
+export interface InputGroupProps extends ComponentPropsWithoutRef<'div'>, InputGroupStylesProps {
+  intent?: InputContainerProps['intent']
   isDisabled?: boolean
 }
 
 export const InputGroup = forwardRef<HTMLDivElement, PropsWithChildren<InputGroupProps>>(
-  ({ className, children: childrenProp, intent = 'neutral', isDisabled, ...others }, ref) => {
-    const [isFocused, setIsFocused] = useState(false)
-    const [isHovered, setIsHovered] = useState(false)
-
+  (
+    { className, children: childrenProp, intent: intentProp = 'neutral', isDisabled, ...others },
+    ref
+  ) => {
+    const { isInvalid } = useFormFieldState()
     const children = Children.toArray(childrenProp).filter(isValidElement)
+    const intent = isInvalid ? 'error' : intentProp
 
     const getDisplayName = (element?: ReactElement) => {
       return element ? (element.type as FC).displayName : ''
     }
 
-    const findElement = (values: string[]) => {
+    const findElement = (...values: string[]) => {
       return children.find(child => values.includes(getDisplayName(child) || ''))
     }
 
-    const input = findElement(['Input', 'TextField'])
-    const left = findElement(['InputGroup.LeftAddon', 'InputGroup.LeftElement'])
-    const right = findElement(['InputGroup.RightAddon', 'InputGroup.RightElement'])
+    const input = findElement('Input', 'TextField')
+    const left = findElement('InputGroup.LeftAddon', 'InputGroup.LeftElement')
```

Comment:
💅, `left` ↔️ `leadingElement`
`right` ↔️ `trailingElement`

Reply:
We will rename everything using those names 🙌🏻